### ISSUE TITLE
Bug in Video captioning generator

### ISSUE THREAD

**Issue by aleSuglia** [VALUE-Leaderboard/StarterCode#7]
Hi @linjieli222,

I was looking at your code for the video captioning generator and I've noticed a potential bug. Specifically, I was looking at the method `greedy_decode` that is reported here: https://github.com/VALUE-Leaderboard/StarterCode/blob/main/model/videoCap.py#L314

At the line https://github.com/VALUE-Leaderboard/StarterCode/blob/main/model/videoCap.py#L338, you compute `output_ids` as the argmax of the logits generated for the current timestep of the decoder. However, it seems that you do not accumulate them in a separate list. Indeed, you reuse the same variable `output_ids` when computing the outputs here: https://github.com/VALUE-Leaderboard/StarterCode/blob/main/model/videoCap.py#L341 

Am I wrong?

Thanks!

**Comment by linjieli222** [comment]
Sorry about the confusion. The [`greedy_decode()`](https://github.com/VALUE-Leaderboard/StarterCode/blob/7f124b4ddef86af887ca593b369d470bb0e3586b/model/videoCap.py#L314) function probably should be written more clearly. Let me explain a little to resolve your concern.

In [`decode()`](https://github.com/VALUE-Leaderboard/StarterCode/blob/7f124b4ddef86af887ca593b369d470bb0e3586b/model/videoCap.py#L254), the `prediction_scores` ( which is the `score` in [`greedy_decode()`](https://github.com/VALUE-Leaderboard/StarterCode/blob/7f124b4ddef86af887ca593b369d470bb0e3586b/model/videoCap.py#L314)) are of size `(N, L, vocab_size)`, where `N` is batch size, and `L` is the current caption length or current decoding steps. 

https://github.com/VALUE-Leaderboard/StarterCode/blob/7f124b4ddef86af887ca593b369d470bb0e3586b/model/videoCap.py#L272-L273

So the `output_ids` will contain all context history, which is of size `(N, L)`.

https://github.com/VALUE-Leaderboard/StarterCode/blob/7f124b4ddef86af887ca593b369d470bb0e3586b/model/videoCap.py#L338



Below is an example I just run for TVC inference:
```
# assuming N = 1
step/L = 1, output_ids = [41415]
step/L = 2, output_ids = [41415,  3026]
step/L = 3, output_ids = [41415,  3026, 12972]
step/L = 4, output_ids = [41415,  3026, 12972, 14]
...
```
Hope this example helps.
